### ISSUE TITLE
Inconsistent datetime parsing 

### ISSUE THREAD

**Issue by michal-kowalczyk** [OpenLEADR/openleadr-python#36]
There's a regex used in parsing datetime which treats microseconds as optional:
https://github.com/OpenLEADR/openleadr-python/blob/main/openleadr/utils.py#L257

When parse_datetime gets as an input something like `2020-12-15T11:29:34Z` it exits with an exception:
```
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

The issue is that microseconds are converted to int no matter if they are present or not:
https://github.com/OpenLEADR/openleadr-python/blob/main/openleadr/utils.py#L259

And if they are absent, the match group value is None, which cannot be converted to int.

**Comment by stan-janssen** [comment]
Thanks, Michal, for reporting this. I just pushed a fixed version to the main branch. Could you check this out and see if it resolves your issue correctly? Thanks!

**Comment by michal-kowalczyk** [comment]
@stan-janssen , I've just verified that fix resolves the issue. Thanks!

**Comment by stan-janssen** [comment]
This fix is included in version `0.5.16`.
